rasa/core/test.py

```diff
@@ -619,11 +627,20 @@ def _collect_action_executed_predictions(
             # that something else than the form was supposed to run.
             predicted = action.name()
 
-    action_executed_eval_store.add_to_store(
-        action_predictions=[predicted], action_targets=[gold]
+    should_ignore_prediction = action_executed_eval_store.should_ignore_prediction(
+        predicted
     )
+    if not should_ignore_prediction:
+        action_executed_eval_store.add_to_store(
+            action_predictions=[predicted], action_targets=[gold]
+        )
 
-    if action_executed_eval_store.has_prediction_target_mismatch():
+    has_prediction_target_mismatch = (
+        action_executed_eval_store.has_prediction_target_mismatch()
+    )
+    if has_prediction_target_mismatch or (
+        should_ignore_prediction and predicted != gold
```

Comment:
I don't totally understand this line, why do we need this?

Reply:
I updated it a bit so I guess it's more clear now.
The idea is simple — we add `WronglyPredictedAction` not only if there is a prediction mismatch but also when `action_unlikely_intent` got predicted.